tests/Unit/ApparentHorizons/Test_Strahlkorper.cpp

```diff
@@ -0,0 +1,426 @@
+    const double sin_phi = sin(phi);
+    const double cos_phi = cos(phi);
+    const double cos_2_theta = cos(2 * theta);
+    const double amp = -sqrt(3.0 / 8.0 / M_PI) * y11_amplitude;
+    test_d2x_radius.get(0, 0)[s] =
+        (9 * sin(3 * phi) * sin_theta -
+         sin_phi * (7 * sin_theta + 12 * square(cos_phi) * sin(3 * theta))) /
+        (16. * square(r));
+    test_d2x_radius.get(0, 1)[s] =
+        -((cos_phi *
+           (square(cos_phi) + ((-1 + 3 * cos_2_theta) * square(sin_phi)) / 2.) *
+           sin_theta) /
+          square(r));
+    test_d2x_radius.get(0, 2)[s] =
+        (3 * cos_phi * cos_theta * sin_phi * square(sin_theta)) / square(r);
+    test_d2x_radius.get(1, 1)[s] =
+        (-3 * (7 * sin_phi * sin_theta + 3 * sin(3 * phi) * sin_theta +
+               4 * pow<3>(sin_phi) * sin(3 * theta))) /
+        (16. * square(r));
+    test_d2x_radius.get(1, 2)[s] =
+        -((cos_theta * (square(cos_phi) +
+                        ((-1 + 3 * cos_2_theta) * square(sin_phi)) / 2.)) /
+          square(r));
+    test_d2x_radius.get(2, 2)[s] =
+        ((1 + 3 * cos_2_theta) * sin_phi * sin_theta) / (2. * square(r));
+    for (auto& a : test_d2x_radius) {
+      a[s] *= amp;
+    }
+  }
+  const auto& strahlkorper_d2x_radius = strahlkorper.d2x_radius();
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      for (size_t s = 0; s < n_pts; ++s) {
+        CHECK(test_d2x_radius.get(i, j)[s] ==
+              approx(strahlkorper_d2x_radius.get(i, j)[s]));
+      }
+    }
+  }
+
+  // Test nabla squared
+  DataVector test_nabla_squared(n_pts);
+  for (size_t s = 0; s < n_pts; ++s) {
+    func.scalar_laplacian(&test_nabla_squared, 1, s, {theta_phi.get(0)[s]},
+                          {theta_phi.get(1)[s]});
+    test_nabla_squared[s] *= y11_amplitude;
+  }
+  const auto& strahlkorper_nabla_squared = strahlkorper.nabla_squared_radius();
+  for (size_t s = 0; s < n_pts; ++s) {
+    CHECK(strahlkorper_nabla_squared[s] == approx(test_nabla_squared[s]));
+  }
+}
+
+void test_normals() {
+  // Create a strahlkorper with a Im(Y11) dependence.
+  const size_t l_max = 4, m_max = 4;
+  const double radius = 2.0;
+  const double y11_amplitude = 1.0;
+  const std::array<double, 3> center = {{0.1, 0.2, 0.3}};
+
+  Strahlkorper<Frame::Inertial> strahlkorper_sphere(l_max, m_max, radius,
+                                                    center);
+  auto coefs = strahlkorper_sphere.coefficients();
+  SpherepackIterator it(l_max, m_max);
+  // Conversion between SPHEREPACK b_lm and real valued harmonic coefficients:
+  // b_lm = (-1)^{m+1} sqrt(1/2pi) d_lm
+  coefs[it.set(1, -1)()] = y11_amplitude * sqrt(0.5 / M_PI);
+  Strahlkorper<Frame::Inertial> strahlkorper(strahlkorper_sphere, coefs);
+
+  const auto& theta_phi = strahlkorper.theta_phi();
+  const auto n_pts = theta_phi.get(0).size();
+
+  // Test surface_tangents
+
+  // Can test_surface_tangents be constructed in one line instead of three?
```

Comment:
Dunno, that's why I asked...

Reply:
Shouldn't the initialiser list constructor work? 
`std::array<Strahlkorper<Frame::Inertial>::ThreeVector, 2> test_surface_tangents{{Strahlkorper<Frame::Inertial>::ThreeVector(n_pts), Strahlkorper<Frame::Inertial>::ThreeVector(n_pts)}};` 

And I'd say that writing `Strahlkorper<Frame::Inertial>::ThreeVector` is more effort than `tnsr::I<DataVector, 3>` since the inertial frame will be defaulted.